components/helm-broker/internal/controller/addonsconfiguration_controller.go

```diff
@@ -0,0 +1,500 @@
+		repo := addons.NewAddonsRepository(specRepository.URL)
+
+		addons, err := r.createAddons(specRepository.URL)
+		if err != nil {
+			repo.Failed()
+			repo.Repository.Reason = addonsv1alpha1.RepositoryURLFetchingError
+			repo.Repository.Message = err.Error()
+			repositories.AddRepository(repo)
+
+			r.log.Errorf("while creating addons for repository from %q: %s", specRepository.URL, err)
+			continue
+		}
+
+		repo.Addons = addons
+		repositories.AddRepository(repo)
+	}
+
+	r.log.Info("- check duplicate ID addons alongside repositories")
+	repositories.ReviseBundleDuplicationInRepository()
+
+	r.log.Info("- check duplicates ID addons in existing addons configuration")
+	list, err := r.existingAddonsConfigurationList(addon)
+	if err != nil {
+		r.log.Errorf("cannot fetch AddonsConfiguration list: %s", err)
+		return exerr.Wrap(err, "while fetching addons configuration list")
+	}
+	repositories.ReviseBundleDuplicationInStorage(list)
+
+	r.log.Info("- save ready bundles and charts in storage")
+	r.saveBundle(internal.Namespace(addon.Namespace), repositories)
+
+	r.log.Info("- update AddonsConfiguration status")
+	r.statusSnapshot(addon, repositories)
+	if err = r.updateAddonStatus(addon); err != nil {
+		r.log.Errorf("cannot update AddonsConfiguration %s: %v", addon.Name, err)
+		return exerr.Wrap(err, "while update AddonsConfiguration status")
+	}
+
+	if err := r.deleteOrphanBundles(addon.Namespace); err != nil {
+		return exerr.Wrap(err, "while deleting orphan bundles from storage")
+	}
+
+	r.log.Info("- ensure ServiceBroker")
+	if err := r.ensureBroker(addon); err != nil {
+		return exerr.Wrap(err, "while ensuring ServiceBroker")
+	}
+
+	return nil
+}
+
+func (r *ReconcileAddonsConfiguration) deleteAddonsProcess(addon *addonsv1alpha1.AddonsConfiguration) error {
+	r.log.Infof("Start delete AddonsConfiguration %s/%s process", addon.Name, addon.Namespace)
+
+	addonsCfgs, err := r.existingAddonsConfigurationList(addon)
+	if err != nil {
+		return exerr.Wrapf(err, "while listing AddonsConfigurations in namespace %s", addon.Namespace)
+	}
+
+	deleteBroker := true
+	for _, addon := range addonsCfgs.Items {
+		if addon.Status.Phase != addonsv1alpha1.AddonsConfigurationReady {
+			// reprocess AddonConfig again if it was failed
+			addon.Spec.ReprocessRequest++
+			if err := r.Client.Update(context.Background(), &addon); err != nil {
+				return exerr.Wrapf(err, "while incrementing a reprocess requests for AddonConfiguration %s/%s", addon.Name, addon.Namespace)
+			}
+		} else {
+			deleteBroker = false
+		}
+	}
+	if deleteBroker {
+		r.log.Info("- delete ServiceBroker from namespace %s", addon.Namespace)
+		if err := r.brokerFacade.Delete(addon.Namespace); err != nil {
+			return exerr.Wrapf(err, "while deleting ServiceBroker from namespace %s", addon.Namespace)
+		}
+	}
+
+	for _, repo := range addon.Status.Repositories {
+		if repo.Status == addonsv1alpha1.RepositoryStatusReady {
```

Comment:
Check status of `AddonsConfiguration` instead each repository

Reply:
done